src/DotNetWorker.Core/Context/FunctionContextHttpRequestExtensions.cs

```diff
@@ -21,16 +22,9 @@ public static class FunctionContextHttpRequestExtensions
         /// <returns>HttpRequestData instance if the invocation is http, else null</returns>
         public static async ValueTask<HttpRequestData?> GetHttpRequestDataAsync(this FunctionContext context)
         {
-            var httpTriggerBinding = context.FunctionDefinition.InputBindings.Values
-                                            .FirstOrDefault(a => a.Type == "httpTrigger");
+            IHttpRequestDataFeature? httpRequestDataFeature = context.Features.Get<IHttpRequestDataFeature>() ?? new DefaultHttpRequestDataFeature();
```

Comment:
I have this now just so the general layout/design can be viewed for this draft but looking to not create a new instance every time.